ext/spl/spl_fixedarray.stub.php

```diff
@@ -49,4 +49,6 @@ public function offsetSet($index, mixed $value) {}
     public function offsetUnset($index) {}
 
     public function getIterator(): Iterator {}
+
+    public function jsonSerialize(): array {}
```

Comment:
This is I think more a question of convention. The `JsonSerializable` interface defines `jsonSerialize()` as having a `mixed` return type. Even though this implementation does only ever return an array, will it cause issues if someone wanted to extend `SplFixedArray` and override `jsonSerialize()` with a different return type like `JsonSerializable`?

I can't think of many situations where there would be a reason someone would have to do this, but I think if they were to be using this stub for intelisense it might complain that the child's method is incompatible with `SplFixedArray`s even though it should technically be allowed by the interface.

This is the deepest I've ever seen into PHPs internals so I don't know how else the stubs are used and whether running that hypothetical example would actually generate any errors.

Reply:
Yes, a child class will not be allowed to widen the type. I guess if we really want we can declare this as a tentative type instead, so that any code hypothetically extending SplFixedArray with a different return type would not be immediately broken.

cc @kocsismate

Reply:
Interesting problem.. Personally, I don't have many too much concern about the `array` return type (I'm not sure how common is to extend `SplFixedArray` + implement `JsonSerializable`), but if we want to be extra safe, a tentative return type seems like a good idea.

Reply:
For what it is worth, I am not overtly concerned either. I just saw it and wasn't sure if it was an overt decision to deviate from the interface.

Reply:
I think I'll go with the explicit type here and change it to a tentative one if anyone complains that it broke their code :) It seems pretty unlikely to me.